src/main/java/net/tropicraft/core/proxy/ClientProxy.java

```diff
@@ -95,6 +103,24 @@ public void registerItemVariantModel(Item item, String name, int metadata) {
 			ModelLoader.setCustomModelResourceLocation(item, metadata, new ModelResourceLocation(Info.MODID + ":" + name, "inventory"));
 		}
 	}
+	
+	private final Map<String, String[]> blockVariants = new HashMap<>();
+	
+	@Override
+	public void registerArbitraryBlockVariants(String name, String... variants) {
+		blockVariants.put(name, variants);
+	}
+	
+	@SubscribeEvent
+	public void onModelBake(ModelBakeEvent event) {
+		for (Entry<String, String[]> e : blockVariants.entrySet()) {
+			for (String variant : e.getValue()) {
+				ModelResourceLocation loc = new ModelResourceLocation(Info.MODID + ":" + e.getKey(), variant);
+				IModel bambooFrame = ModelLoaderRegistry.getModelOrLogError(loc, "Could not load arbitrary block variant " + variant + " for block " + e.getKey());
```

Comment:
If this is a generic object we should probably call it something more generic than bambooFrame 😄